### ISSUE TITLE
TextInput widget won't accept enter in Safari

### ISSUE THREAD

**Issue by NaserMonsefi** [bokeh/bokeh#6719]
Hi

I have been testing Bokeh (0.12.6) for a small dashboards and noticed that TextInput widget won't pass any inputs in Internet Explorer 11 or Safari when enter is pressed. The dashboard works fine in chrome. I added a submit button for now but not sure if this is something that can be addressed. My python is 3.5.3.

I also noticed that even in the demo [Movies Explorer](https://demo.bokehplots.com/apps/movies) same problem exists. Making both text input fields completely useless. 

![image](https://user-images.githubusercontent.com/13728402/28958933-408f8cc4-78f0-11e7-837f-d910df513be5.png)

Will this issue be address any time soon?

Thanks for the great work,
Yours,

**Comment by bryevdv** [comment]
@NaserMonsefi I use the text input widget in Safari very often without issue. Can you provide more detailed version and system configuration information to reproduce?

**Comment by NaserMonsefi** [comment]
Thanks for your message @bryevdv . Unfortunately I don't have access to the my own code at the moment but will share the details soon. In short I used Bokeh 0.12.6 on Python 3.5.3.
But even for the [Movies Explorer demo](https://demo.bokehplots.com/apps/movies) neither **IE 11.483.15063.0** nor Safari **10.1.2 (12603.3.8)** works with the TextInput.

**Comment by bryevdv** [comment]
@NaserMonsefi  I am not sure what to suggest. The Bokeh version on the demo site is a little older than `0.12.6` (I have not had an opportunity to redeploy) but regardless the text input works fine for me just now with Safari Version 10.1.2 (12603.3.8).

Is there any output in the browser's JS console log?

**Comment by NaserMonsefi** [comment]
@bryevdv , I am trying to use Bokeh for some dashboards to be shared within our company that mostly uses IE. I get two warnings on the IE console (attached here) that I don't get in Chrome.
![image](https://user-images.githubusercontent.com/13728402/29005600-50b0d6a2-7ad6-11e7-96ec-fa2b7c848a2f.png)
I am not sure if that is any help?

**Comment by bryevdv** [comment]
Is that only after the  page load, or after trying to use the text input?

**Comment by NaserMonsefi** [comment]
That's after the page got loaded. Typing in the text field and pressing enter won't do anything. Only the "X" appears in the corner of text input.

**Comment by bryevdv** [comment]
@NaserMonsefi I just tried this on IE end Edge on Win10. It works as expected on Edge. On IE, it works, but not after hitting enter, it only registers if I change focus to another widget. Can you confirm whether you see that same or not?

**Comment by NaserMonsefi** [comment]
Yes, That is the exact thing I see as well. It works fine on Edge, On IE it only register the text input after I move one of the sliders.

**Comment by bryevdv** [comment]
OK well that's something, perhaps there is some different MS specific input event t listen for @mattpap any ideas?

I am not sure what to suggest for Safari. Safari is my daily browser, I use it constantly and I can't reproduce any issue, which will make investigating things difficult/impossible.

**Comment by NaserMonsefi** [comment]
I see, I will be more focused on IE anyways. I added a submit button to my dashboard which works around this but for only dashboards with only text input this would be a bigger issue.

**Comment by bryevdv** [comment]
OK well hopefully it will be a simple fix, I've triaged for 0.12.8 for now since you have a workaround for the time being.

**Comment by NaserMonsefi** [comment]
@bryevdv Hopefully it is a simple one. Great, Thanks.

**Comment by bryevdv** [comment]
just an update with the 1.1dev build deployed to `demo.bokeh.org` it still does not submit on enter, but does submit on tab. This SO answer may have some pointers 

https://stackoverflow.com/questions/4715249/submit-form-on-enter-in-ie-web-browser

**Comment by mattpap** [comment]
@bokeh/dev, can anyone, who has access to Safari, confirm that this is still an issue? If not reproducible, then we can close this as IE compatibility is not a concern anymore.

**Comment by IuryPiva** [comment]
Just tried on Safari, everything works as expected

**Comment by mattpap** [comment]
OK, perhaps this was an issue with an older version of Safari.
